src/main/remote-sync/RemoteSyncManager.ts

```diff
@@ -0,0 +1,347 @@
+import Logger from 'electron-log';
+import * as helpers from './helpers';
+import {
+  RemoteSyncStatus,
+  RemoteSyncedFolder,
+  RemoteSyncedFile,
+  SyncConfig,
+  SYNC_OFFSET_MS,
+} from './helpers';
+import { reportError } from '../bug-report/service';
+import { obtainToken } from '../auth/service';
+import { DatabaseAdapter } from '../database/adapters/base';
+
+export class RemoteSyncManager {
+  private foldersSyncStatus: RemoteSyncStatus = 'IDLE';
+  private filesSyncStatus: RemoteSyncStatus = 'IDLE';
+  private status: RemoteSyncStatus = 'IDLE';
+  private onStatusChangeCallback: (newStatus: RemoteSyncStatus) => void =
+    () => {
+      return;
+    };
+  constructor(
+    private db: {
+      files: DatabaseAdapter<RemoteSyncedFile>;
+      folders: DatabaseAdapter<RemoteSyncedFolder>;
+    },
+    private config: {
+      fetchFilesLimitPerRequest: number;
+      fetchFoldersLimitPerRequest: number;
+      syncFiles: boolean;
+      syncFolders: boolean;
+    }
+  ) {}
+
+  onStatusChange(callback: (newStatus: RemoteSyncStatus) => void) {
```

Comment:
Could be useful to use an array to store the listeners of this event in case they are multiple? Replacing it could be a bit confusing.

Using an `EventEmitter` could provide this way of working by default (like extending it)

Reply:
+1 to have an event emitter, but I would prefer to have it injected rather than extending from it.

Reply:
@JoanVicens  what do you mean by injected rather than extending from it?

Reply:
Now callbacks are stored in an array so multiple callbacks can be passed

Reply:
I think @JoanVicens is referring to injecting the `EventEmitter` via constructor @PixoDev

Reply:
ok, but why? I added a simple callback pattern here (now with arrays we can handle N callbacks) because this is the only method that uses callbacks in this class, maybe in the future we add more stuff that needs a more sophisticated system, but I believe the callbacks pattern gets the job done here